### PR TITLE
Added on_close callback to client.rb

### PR DESCRIPTION
**Pull request by rayning0**
The em-twitter gem lets us check for a closed connection, but tweetstream does not. I fixed it.

Now I can check when Twitter connection closes. Tied to the 'close_callback' and 'on_close' method in the em-twitter gem's client.rb, plus to 'unbind' method in its connection.rb.

### CODE REVIEW

**Comment by sferik** [comment]
I’ve fixed the build in `master`. Please rebase your branch and all the RuboCop offenses should go away.

File: lib/tweetstream/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module TweetStream
-  VERSION = '2.6.1'
+  VERSION = '2.6.2'
```
Comment (stve) [inline lib/tweetstream/version.rb]: please revert this change, we prefer to change this only when cutting releases, thanks

File: spec/tweetstream/client_authentication_spec.rb [hunk 2]
```diff
@@ -2,17 +2,19 @@
...
+      :unbind => true,
+      :each => true,
+      :on_error => true,
+      :on_max_reconnects => true,
+      :on_reconnect => true,
+      :connection_completed => true,
+      :on_no_data_received => true,
+      :on_unauthorized => true,
+      :on_enhance_your_calm => true,
+      :on_close => true,
```
Comment (stve) [inline spec/tweetstream/client_authentication_spec.rb]: there's a few diffs with changes like this, this just adds diff-noise to the commit, if possible, can you please revert this and the other instances of indentation changes?

Comment (stve) [inline spec/tweetstream/client_authentication_spec.rb]: oh, looks like these were based on rubocop barking, all good :smile_cat:

Comment (rayning0) [inline spec/tweetstream/client_authentication_spec.rb]: RuboCop forced me to do this. I got RuboCop offenses for each @stream = double because RuboCop wanted the ending ")" to be in same column as starting "(". It wouldn't accept my code. After I changed all that, @sferik  changed the master branch to have all the indents go the way they are now.

**Comment by stve** [comment]
@rayning0 thanks for the pull request. Can you squash your commits and we'll get this merged. Thanks.

**Comment by rayning0** [comment]
Ok @stve and @sferik, I just squashed all my commits and changed the gem version back to it's original 3. Please merge my code.

**Comment by rayning0** [comment]
Ok @stve and @sferik, I squashed all my commits and changed the gem version back. Please MERGE!

**Comment by sferik** [comment]
@rayning0 Thanks!
